frontend/src/components/common/FilterBox.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component, PropTypes } from 'react'
+
+import onClickOutsideAddon from 'react-onclickoutside'
+
+class FilterBox extends Component {
+  static propTypes = {
+    topic: PropTypes.object,
+    map: PropTypes.object,
+    filterData: PropTypes.object,
+    allForFiltering: PropTypes.object,
+    visibleForFiltering: PropTypes.object,
+    toggleMetacode: PropTypes.func,
+    toggleMapper: PropTypes.func,
+    toggleSynapse: PropTypes.func,
+    filterAllMetacodes: PropTypes.func,
+    filterAllMappers: PropTypes.func,
+    filterAllSynapses: PropTypes.func,
+    closeBox: PropTypes.func
+  }
+
+  handleClickOutside = () => {
+    this.props.closeBox()
+  }
+
+  render () {
+    const { topic, map, filterData, allForFiltering, visibleForFiltering, toggleMetacode,
+            toggleMapper, toggleSynapse, filterAllMetacodes,
+            filterAllMappers, filterAllSynapses } = this.props
+    const style = {
+      maxHeight: document.body.clientHeight - 108 + 'px'
+    }
+    const mapperAllClass = "showAll showAllMappers"
+                              + (allForFiltering.mappers.length === visibleForFiltering.mappers.length ? ' active' : '')
+    const mapperNoneClass = "hideAll hideAllMappers"
+                              + (visibleForFiltering.mappers.length === 0 ? ' active' : '')
+    const metacodeAllClass = "showAll showAllMetacodes"
+                              + (allForFiltering.metacodes.length === visibleForFiltering.metacodes.length ? ' active' : '')
+    const metacodeNoneClass = "hideAll hideAllMetacodes"
+                              + (visibleForFiltering.metacodes.length === 0 ? ' active' : '')
+    const synapseAllClass = "showAll showAllSynapses"
+                              + (allForFiltering.synapses.length === visibleForFiltering.synapses.length ? ' active' : '')
+    const synapseNoneClass = "hideAll hideAllSynapses"
+                              + (visibleForFiltering.synapses.length === 0 ? ' active' : '')
+    return map || topic ? <div className="sidebarFilterBox upperRightBox" style={style}>
+      <div className="filterBox">
+        <h2>FILTER BY</h2>
+        <div id="filter_by_mapper" className="filterBySection">
+          {map && <h3>MAPPERS</h3>}
+          {topic && <h3>CREATORS</h3>}
+          <span className={mapperNoneClass} onClick={() => filterAllMappers()}>NONE</span>
+          <span className={mapperAllClass} onClick={() => filterAllMappers(true)}>ALL</span>
+          <div className="clearfloat"></div>
+          <ul>
+            {allForFiltering.mappers.map(m => {
+              const data = filterData.mappers[m]
+              const isVisible = visibleForFiltering.mappers.indexOf(m) > -1
```

Comment:
Can you use mapper instead of m? 

i.e. `const data = filterData.mappers[mapper]`

Or are these ids? Maybe describing if it's a mapperName or a mapperId would help. I'm not even sure what it is as I read this